### PR TITLE
Warn about overriding parameters while reading the job .ini file

### PR DESCRIPTION
**Pull request by ptormene**
Fixes https://github.com/gem/oq-engine/issues/8503

Since `configparser` considers it reasonable to define a parameter in a default section, so it does not need to be defined in the section it would naturally belong to, or it can be inserted in that section to override the default, I am assuming that the engine should follow a similar paradigm, without raising an error in case a parameter is present in two different sections. However, I've added a check that will print a warning in case any of such re-definitions are found.

### CODE REVIEW

File: openquake/commonlib/readinput.py [hunk 1]
```diff
@@ -228,6 +228,20 @@ def _update(params, items, base_path):
...
 
 
+def _warn_about_duplicates(cp):
+    sections = cp.sections()
+    prev_sections = []
+    for curr_sect in sections:
+        prev_sections.append(curr_sect)
+        other_sects = [sect for sect in sections if sect not in prev_sections]
+        for key in dict(cp.items(curr_sect)).keys():
+            for other_sect in other_sects:
```
Comment (micheles) [inline openquake/commonlib/readinput.py:238]: You don't need `.keys()`

File: openquake/commonlib/readinput.py [hunk 2]
```diff
@@ -228,6 +228,20 @@ def _update(params, items, base_path):
...
+                if key in dict(cp.items(other_sect)).keys():
```
Comment (micheles) [inline openquake/commonlib/readinput.py:239]: You don't need `.keys()`

File: openquake/commonlib/readinput.py (hunk 1 above)
Comment (ptormene) [reply openquake/commonlib/readinput.py:238]: done

File: openquake/commonlib/readinput.py (hunk 2 above)
Comment (ptormene) [reply openquake/commonlib/readinput.py]: done

**Review by micheles — APPROVED** [review]

File: openquake/commonlib/readinput.py [hunk 3]
```diff
@@ -228,6 +228,20 @@ def _update(params, items, base_path):
             params[key] = value
 
 
+def _warn_about_duplicates(cp):
+    sections = cp.sections()
+    prev_sections = []
+    for curr_sect in sections:
+        prev_sections.append(curr_sect)
+        other_sects = [sect for sect in sections if sect not in prev_sections]
+        for key in dict(cp.items(curr_sect)):
```
Comment (micheles) [inline openquake/commonlib/readinput.py:237]: Why not simply `for key, _ in cp.items(curr_sect)`?

File: openquake/commonlib/readinput.py [hunk 4]
```diff
@@ -228,6 +228,20 @@ def _update(params, items, base_path):
             params[key] = value
 
 
+def _warn_about_duplicates(cp):
+    sections = cp.sections()
+    prev_sections = []
+    for curr_sect in sections:
+        prev_sections.append(curr_sect)
+        other_sects = [sect for sect in sections if sect not in prev_sections]
```
Comment (micheles) [inline openquake/commonlib/readinput.py:236]: This algorithm is quadratic and really wrong to use in general (one should use sets) however configuration files has extremely few sections so it is a non-issue here.

**Pull request merged**